risks/MASVS-STORAGE/1-store-sensitive-data-securely/data-unencrypted-external/android-data-unencrypted-external-frida/example-1/snippet.kt

```diff
@@ -0,0 +1,7 @@
+// On lower Android versions, you might need to add `WRITE_EXTERNAL_STORAGE` permission to the manifest to write to an external app-specific directory.
+
+val externalDirPath = getExternalFilesDir(null)!!.absolutePath
+val file: File = File("$externalDirPath/secret.json")
+FileOutputStream(file).use { fos ->
+    fos.write("password:123".toByteArray())
+}
```

Comment:
Maybe better include the comment in the demo.md

Here's the new file using the new app:

```suggestion
package org.owasp.mastestapp

import android.content.Context
import android.os.Environment
import android.os.Environment.getExternalStoragePublicDirectory
import java.io.File
import java.io.FileOutputStream

class MastgTest (private val context: Context){

    fun mastgTest(): String {
        val secret1 = "{\"password\":\"12345\"}\n"
        val externalDirPath = getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)
        val file1 = File("$externalDirPath/secrets.json")
        FileOutputStream(file1).use { fos ->
            fos.write(secret1.toByteArray())
        }

        return "Secrets written to $file1:\n\n$secret1"
    }

}
```